src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int OPERATION_AMOUNT_INDEX = 1;
+    private static final String BUY_OPERATION = "buy";
+    private static final String SUPPLY_OPERATION = "supply";
+    private static final String SPLIT_REGEX = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int buySum = countSupplyOrBuyInFile(fromFileName, BUY_OPERATION);
+        int supplySum = countSupplyOrBuyInFile(fromFileName, SUPPLY_OPERATION);
+        int result = supplySum - buySum;
+
+        try (BufferedWriter output = new BufferedWriter(new FileWriter(toFileName))) {
+            StringBuilder builder = new StringBuilder();
+            builder.append("supply").append(",").append(supplySum).append(System.lineSeparator())
+                    .append("buy").append(",").append(buySum).append(System.lineSeparator())
+                    .append("result").append(",").append(result);
+            output.write(builder.toString());
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot write file " + e);
+        }
+    }
+
+    private int countSupplyOrBuyInFile(String fromFileName, String option) {
+        if (!option.equals(BUY_OPERATION) && !option.equals(SUPPLY_OPERATION)) {
+            throw new WrongOptionException("Wrong option");
+        }
 
+        try (BufferedReader input = new BufferedReader(new FileReader(fromFileName))) {
+            String value = input.readLine();
+            int supplyCounterReturn = 0;
```

Comment:
The variable name 'supplyCounterReturn' does not accurately describe its purpose, as it counts both supply and buy operations depending on the option. Consider renaming it to 'operationSum' or a similar name that reflects its functionality.